File: src/main/java/fr/vlegall/sochief/service/category/RecipeCategoryService.java
package fr.vlegall.sochief.service.category;

import fr.vlegall.sochief.contracts.common.NamedIdDto;
import fr.vlegall.sochief.repository.RecipeCategoryRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
public class RecipeCategoryService implements IRecipeCategoryService {
    private final RecipeCategoryMapper recipeCategoryMapper;
    private final RecipeCategoryRepository recipeCategoryRepository;

    public RecipeCategoryService(RecipeCategoryMapper recipeCategoryMapper, RecipeCategoryRepository recipeCategoryRepository) {
        this.recipeCategoryMapper = recipeCategoryMapper;
        this.recipeCategoryRepository = recipeCategoryRepository;
    }

    @Override
    public List<NamedIdDto> getRecipeCategories() {
        return recipeCategoryRepository.findAllByOrderByIdAsc()
                .stream()
                .map(recipeCategoryMapper::toDto)
                .collect(Collectors.toList());
    }
}

File: src/main/kotlin/fr/vlegall/sochief/service/category/RecipeCategoryService.kt
package fr.vlegall.sochief.service.category

import fr.vlegall.sochief.repository.RecipeCategoryRepository
import fr.vlegall.sochief.contracts.common.NamedIdDto
import org.springframework.stereotype.Service
import kotlin.collections.map

@Service
class RecipeCategoryService(
    private val recipeCategoryMapper: RecipeCategoryMapper,
    private val recipeCategoryRepository: RecipeCategoryRepository
) : IRecipeCategoryService {
    override fun getRecipeCategories(): List<NamedIdDto> {
        return recipeCategoryRepository.findAllByOrderByIdAsc().map(recipeCategoryMapper::toDto)
    }

}